packages/atomic/src/utils/store.ts

```diff
@@ -32,6 +32,13 @@ export type AtomicStore = {
   facetElements: HTMLElement[];
   sortOptions: SortDropdownOption[];
   iconAssetsPath: string;
+  /**
+   * Dynamic list of flags that allows for syncing various components loading state on initialization.
+   * E.g., waiting for result template component to be rendered.
+   *
+   * This is a better indicator than only the "firstSearchExecuted" of the Headless search state.
+   */
+  loadingFlags: string[];
```

Comment:
Key addition of this PR, I started with only a "firstResultLoaded" flag but I realized it wouldn't scale like support smart snippets (also have a delayed rendering) or multiple result lists. Atomic users would potentially be able to add flags of their own, to sync nicely with the rendering of the result list.

I decided not to type for that reason:
- allows users to add new ones
- multiple flags for multiple result lists

Reply:
> Atomic users would potentially be able to add flags of their own, to sync nicely with the rendering of the result list.

I think that's just "in theory" though.

If we start with typed flags that we only use internally, then nothing stop us from expanding and making it public down the line if it really becomes a need. So I would vote to keep this private/typed/internal only, and then open it up later if needed.

VS starting with it being public, and it's harder to change/modify. What do you think ?

Reply:
> multiple flags for multiple result lists
This is used right now, each list has it's unique flag ID. The smart snippet could also be added here...
If we type the flags we might not be able to support this